compiler+runtime/src/cpp/jank/runtime/core/seq.cpp

```diff
@@ -790,6 +790,10 @@ namespace jank::runtime
         }
         else if constexpr(behavior::seqable<T>)
         {
+          if constexpr(!behavior::sequential<T> && !std::is_same_v<T, obj::persistent_string>)
+          {
+            throw std::runtime_error{ fmt::format("not ordered: {}", object_type_str(o->type)) };
+          }
           native_integer i{};
```

Comment:
I wonder if this should be `size_t`?

Reply:
This is a `native_integer` because `index` is. Comparing a `size_t` to an `native_integer` would lead to a compiler error. Since the incoming index is a `native_integer`, it's impossible for it to represent beyond the range of the type, so the `size_t` doesn't buy us anything aside from some consistency, at the cost of requiring us to cast `index` to a local `size_t` after verifying it's `>= 0`.